api/src/main/java/net/kyori/adventure/bossbar/BossBar.java

```diff
@@ -52,7 +53,6 @@
    * @throws IllegalArgumentException if percent is less than 0 or greater than 1
    */
   static @NonNull BossBar of(final @NonNull Component name, final float percent, final @NonNull Color color, final @NonNull Overlay overlay) {
-    BossBarImpl.checkPercent(percent);
```

Comment:
This is intentional, checks before attempting to create it this way.

Reply:
meaning other ways of bossbar creation may have out-of-range percentage?

Reply:
Having a check inside the constructor seems reasonable? Also what liach said.

Reply:
Check inside constructor is bad if out-of-range percentages may exist.

Reply:
I don't follow